### PR TITLE
Create viewed slider

### PR DESCRIPTION
**Pull request by WhitShake**
Added "Viewed" slider to mark files as read in APIView.

### CODE REVIEW

**Review by chidozieononiwu — APPROVED** [review]
Doesn't look like you added all the changes in the PR.

**Comment by WhitShake** [comment]
> You did not add all the files

No idea how I pulled that off. I did a new push and it looks like they're all there now, but let me know.

File: src/dotnet/APIView/APIViewWeb/LeanModels/ReviewListModels.cs [hunk 1]
```diff
@@ -128,6 +128,7 @@ public class APIRevisionListItemModel : BaseListitemModel
         public bool IsDeleted { get; set; }
         public bool IsReleased { get; set; }
         public DateTime ReleasedOn { get; set; }
+        public HashSet<string> ViewedUsers { get; set; } = new HashSet<string>();
```
Comment (chidozieononiwu) [inline src/dotnet/APIView/APIViewWeb/LeanModels/ReviewListModels.cs:131]: Can you name this `ViewedBy`

File: src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml [hunk 2]
```diff
@@ -475,7 +475,37 @@
                     </div>
                 </li>
             }
-        </ul>             
+        </ul>
+        <p class="h6">
+            <a data-bs-toggle="collapse" href="#apiRevisionOptionsCollapse" aria-expanded="true" aria-controls="approvalCollapse">API Revision Options&nbsp;&nbsp;<i class="fa-solid fa-ellipsis"></i></a>
```
Comment (chidozieononiwu) [inline src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml:480]: Can you change aria-controls value to `apiRevisionOptionsCollapse`. aria-controld value should be the same as the href without the '#'

Could you also update these lines
https://github.com/Azure/azure-sdk-tools/blob/115b3f0f30b4a96d15567ab03bee0b4f7354957a/src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml#L334

https://github.com/Azure/azure-sdk-tools/blob/115b3f0f30b4a96d15567ab03bee0b4f7354957a/src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml#L369

File: src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml [hunk 3]
```diff
@@ -475,7 +475,37 @@
...
                 </li>
             }
-        </ul>             
+        </ul>
+        <p class="h6">
+            <a data-bs-toggle="collapse" href="#apiRevisionOptionsCollapse" aria-expanded="true" aria-controls="approvalCollapse">API Revision Options&nbsp;&nbsp;<i class="fa-solid fa-ellipsis"></i></a>
+        </p>
+        @{
+            var apiRevisionOptionsCollapseState = " show";
+            if (Request.Cookies.ContainsKey("APIRevisionOptionsCollapse"))
```
Comment (chidozieononiwu) [inline src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml:484]: Cookie value is case sensitive. Ensure it is the same everywhere you rare using it.

File: src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml [hunk 4]
```diff
@@ -475,7 +475,37 @@
...
+            {
+                if (!Request.Cookies["apiRevisionOptionsCollapse"].Equals("shown"))
```
Comment (chidozieononiwu) [inline src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml:486]: Ensure the cookie name is the same everywhere you are using it.

File: src/dotnet/APIView/APIViewWeb/Client/src/pages/review.ts [hunk 5]
```diff
@@ -193,14 +193,18 @@ $(() => {
...
+  });
   // Toggle Close Switch
   $("#reviewCloseSwitch").on('change', function () {
     $("#reviewCloseForm").submit();
   });
 
   // Manage Expand / Collapse State of options
   [$("#approveCollapse"), $("#requestReviewersCollapse"), $("#reviewOptionsCollapse"), $("#pageSettingsCollapse"),
-    $("#associatedPRCollapse"), $("#associatedReviewsCollapse"), $("#generateAIReviewCollapse")].forEach(function (value, index) {
+    $("#associatedPRCollapse"), $("#associatedReviewsCollapse"), $("#generateAIReviewCollapse"), $("#apiRevisionOptionsCollapse")].forEach(function (value, index) {
```
Comment (chidozieononiwu) [inline src/dotnet/APIView/APIViewWeb/Client/src/pages/review.ts:207]: Cookie name used here should match everywhere else. Cookie name is case sensitive.

File: src/dotnet/APIView/APIViewWeb/LeanModels/ReviewListModels.cs (hunk 1 above)
Comment (WhitShake) [reply src/dotnet/APIView/APIViewWeb/LeanModels/ReviewListModels.cs:131]: Changed this one and all references

File: src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml (hunk 3 above)
Comment (WhitShake) [reply src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml]: All changes made.

File: src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml (hunk 2 above)
Comment (WhitShake) [reply src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml]: Made 3 changes

**Review by chidozieononiwu — APPROVED** [review]

**Pull request merged**